clang/lib/Sema/SemaExprCXX.cpp

```diff
@@ -4843,7 +4843,7 @@ Sema::PerformImplicitConversion(Expr *From, QualType ToType,
                  .get();
       break;
     case ICK_Floating_Integral:
-      if (ToType->isRealFloatingType())
+      if (ToType->isRealFloatingType() || ToType->hasFloatingRepresentation())
```

Comment:
I think `isRealFloatingType` is redundant now. `hasFloatingRepresentation` will return true in all cases that it does. I guess this also means that complex types will go into this case now and they didn't before - does that matter?

Reply:
I can fix that up. 
For your second question line 4800  `assert(From->getType()->isVectorType() && ToType->isVectorType()`. It should just be vectors for now.